### PR TITLE
Update all admin*.js to ES6 #6785

### LINKED ISSUE #6785: Update all admin*.js to ES6
Related to #6343, updating all admin related files to es6.

Similar: #6767, #6771 , #6772 , #6773 .

### PR DESCRIPTION
**Pull request by li-kai**
Fixes #6785 

**Outline of Solution**

Most of these are solved by `--fix`, but there also some other changes.

### CODE REVIEW

File: src/main/webapp/js/adminAccountManagement.es6 [hunk 1]
```diff
@@ -89,18 +83,11 @@ function showFirstPage() {
...
-}
-
 function reLabelOrderedAccountEntries() {
     total = 0;
-    $('tr.accountEntry').each(function(index) {
-        $(this).attr('id', 'accountEntry_' + (index + 1));
-        total++;
+    $('tr.accountEntry').each(function (index) {
+        $(this).attr('id', `accountEntry_${index + 1}`);
+        total += 1;
```
Comment (Neurrone) [inline src/main/webapp/js/adminAccountManagement.es6]: Why the change to +=?

File: src/main/webapp/js/adminAccountManagement.es6 [hunk 2]
```diff
@@ -145,18 +116,18 @@ function goToNextPage() {
 }
 
 function bindDeleteAccountAction() {
-    $('.admin-delete-account-link').on('click', function(event) {
+    $('.admin-delete-account-link').on('click', (event) => {
```
Comment (Neurrone) [inline src/main/webapp/js/adminAccountManagement.es6]: Are we standardizing on normal or arrow functions in jquery methods?

File: src/main/webapp/js/adminAccountManagement.es6 [hunk 3]
```diff
@@ -145,18 +116,18 @@ function goToNextPage() {
...
         event.preventDefault();
 
-        var $clickedLink = $(event.target);
-        var googleId = $clickedLink.data('googleId');
-        var existingCourses = document.getElementById('courses_' + googleId).innerHTML;
+        const $clickedLink = $(event.target);
+        const googleId = $clickedLink.data('googleId');
+        const existingCourses = document.getElementById(`courses_${googleId}`).innerHTML;
```
Comment (Neurrone) [inline src/main/webapp/js/adminAccountManagement.es6]: Replace with jquery API?

File: src/main/webapp/js/adminEmail.es6 [hunk 4]
```diff
@@ -1,81 +1,52 @@
-'use strict';
+/* global StatusType:false setStatusMessage:false toggleSort:false richTextEditorBuilder:false */
 
 // Form input placeholders
-var PLACEHOLDER_IMAGE_UPLOAD_ALT_TEXT = 'Please enter an alt text for the image';
+const PLACEHOLDER_IMAGE_UPLOAD_ALT_TEXT = 'Please enter an alt text for the image';
```
Comment (Neurrone) [inline src/main/webapp/js/adminEmail.es6]: Move to const.js?

File: src/main/webapp/js/adminEmail.es6 [hunk 5]
```diff
@@ -1,81 +1,52 @@
...
+            setTimeout(() => {
                 if (data.isError) {
                     setErrorMessage(data.ajaxStatus);
                 } else {
                     $('#adminEmailReceiverListForm').attr('action', data.nextUploadUrl);
                     setStatusMessage(data.ajaxStatus);
-                    submitGroupReceiverListUploadFormAjax();
+                    submitGroupReceiverListUploadFormAjax();    // eslint-disable-line no-use-before-define
                 }
             }, 500);
```
Comment (Neurrone) [inline src/main/webapp/js/adminEmail.es6]: Not sure if its a good idea to remove the 500 ms delay here and in other places? It probably won't break anything, but maybe might be better to minimize functionality changes.

File: src/main/webapp/js/adminAccountManagement.es6 (hunk 1 above)
Comment (li-kai) [inline src/main/webapp/js/adminAccountManagement.es6]: https://github.com/airbnb/javascript#variables--unary-increment-decrement

File: src/main/webapp/js/adminAccountManagement.es6 (hunk 2 above)
Comment (li-kai) [inline src/main/webapp/js/adminAccountManagement.es6]: if there are no usage of `this` they can be safely converted to arrow functions

**Comment by Neurrone** [comment]
@li-kai approving; lets delay those to a different PR per Slack.

**Comment by wkurniawan07** [comment]
I'll help de-conflicting and re-ordering some of the functions, as per what I did in #6790.

File: src/main/webapp/js/adminAccountManagement.es6 [hunk 6]
```diff
@@ -109,24 +108,23 @@ function reLabelOrderedAccountEntries() {
 }
 
 function showEntriesForSelectedPage() {
-    begin = (currentPage - 1) * entryPerPage + 1;
-    end = begin + entryPerPage - 1;
+    begin = (currentPage - 1) * (entryPerPage + 1);
```
Comment (wkurniawan07) [inline src/main/webapp/js/adminAccountManagement.es6]: Doesn't this change the logic?

(although admittedly I don't know where this is used)

File: src/main/webapp/js/adminEmail.es6 [hunk 7]
```diff
@@ -1,81 +1,78 @@
...
-            setTimeout(function() {
+        success(data) {
+            setTimeout(() => {
                 if (data.isError) {
                     setErrorMessage(data.ajaxStatus);
                 } else {
                     $('#adminEmailReceiverListForm').attr('action', data.nextUploadUrl);
                     setStatusMessage(data.ajaxStatus);
-                    submitGroupReceiverListUploadFormAjax();
+                    submitGroupReceiverListUploadFormAjax();    // eslint-disable-line no-use-before-define
```
Comment (wkurniawan07) [inline src/main/webapp/js/adminEmail.es6]: Can undo the changes here.

File: src/main/webapp/js/adminEmail.es6 [hunk 8]
```diff
@@ -107,41 +104,39 @@ function submitGroupReceiverListUploadFormAjax() {
...
-            setTimeout(function() {
+        success(data) {
+            setTimeout(() => {
                 if (data.isError) {
                     setErrorMessage(data.ajaxStatus);
                 } else {
                     $('#adminEmailFileForm').attr('action', data.nextUploadUrl);
                     setStatusMessage(data.ajaxStatus);
-                    submitImageUploadFormAjax();
+                    submitImageUploadFormAjax();        // eslint-disable-line no-use-before-define
```
Comment (wkurniawan07) [inline src/main/webapp/js/adminEmail.es6]: Here as well.

File: src/main/webapp/js/adminHome.es6 [hunk 9]
```diff
@@ -73,21 +69,21 @@ function addInstructorAjax(isError, data) {
...
         $('#addInstructorDetailsSingleLine').val(instructorsToBeRetried);
     }
-    paramsCounter++;
-    var panelHeader = '<strong>Result (' + paramsCounter + '/' + paramsList.length + ')</strong>';
+    paramsCounter += 1;
+    const panelHeader = `<strong>Result (${paramsCounter}/${paramsList.length})</strong>`;
     $('#addInstructorResultPanel div.panel-heading').html(panelHeader);
     if (paramsCounter < paramsList.length) {
-        addInstructorByAjaxRecursively();
+        addInstructorByAjaxRecursively();       // eslint-disable-line no-use-before-define
```
Comment (wkurniawan07) [inline src/main/webapp/js/adminHome.es6]: Here as well.
